File: solution_set1/1523. Count Odd Numbers in an Interval Range.py
##ss
class Solution:
    def countOdds(self, low: int, high: int) -> int:
        
        ##count of odd numbers
        
        if low%2==0 and high%2==0:
            return (high - low)//2
        
        elif low%2!=0 and high%2==0:
            return 1 + (high-low-1)//2
        
        elif low%2==0 and high%2!=0:
            return 1 + (high-1-low)//2
        
        else:
            return 2 + (high-low-2)//2
        

File: cpp_solutions/1523. Count Odd Numbers in an Interval Range.cpp
class Solution {
public:
    int countOdds(int low, int high) {

        if(low%2==0 and high%2==0){
            return (int)(high-low)/2;
        }
        else if(low%2==0 and high%2!=0){
            return (int)((high-1-low)/2) + 1;
        }
        else if(low%2!=0 and high%2==0){
            return (int)((high-low-1)/2) + 1;
        }
        else{
            return (int)((high-low)/2) + 1;
        }
        
    }
};
